modules/@apostrophecms/schema/ui/apos/logic/AposInputArray.js

```diff
@@ -247,14 +371,47 @@ export default {
       const item = this.items.find(item => item._id === id);
       return get(item.schemaInput.data, titleField) || `Item ${index + 1}`;
     },
-    openInlineItem(id) {
-      this.items.forEach(item => {
-        item.open = (item._id === id) || this.alwaysExpand;
-      });
+    toggleOpenInlineItem(event) {
+      if (event) {
+        const el =
+          event.target.getAttribute('data-id')
+            ? event.target
+            : event.target.closest('[data-id]');
+        const elId = el.getAttribute('data-id');
+        if (this.items.find(i => i._id === elId)) {
+          event.preventDefault();
+          const was = this.items.find(i => i._id === elId).open;
+          this.items.find(i => i._id === elId).open = !was;
+        }
+      }
```

Comment:
early  return to avoid nesting and can run the `find` only once instead of three times:
```javascript
    toggleOpenInlineItem(event) {
      if (event) {
        return;
      }
      const el = event.target.getAttribute('data-id')
        ? event.target
        : event.target.closest('[data-id]');
      const elId = el.getAttribute('data-id');
      const elItem = this.items.find(({ _id }) => ._id === elId);
      if (elItem) {
        event.preventDefault();
        elItem.open = !elItem.open;
      }
    }
```